### PR TITLE
Update Demos for latest Featuretools and EvalML versions

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
This PR updates all the demos to work with the latest versions of Featuretools (1.3.0) and EvalML (0.39.0).

Separate PRs will be submitted against the `ft1.0-update` branch to keep PR reviews more manageable. Once all notebooks have been updated, this PR can be merged.

**Notebooks to Update:**
- [x] Update `predict-appointment-noshow`
- [x] Update `predict-correct-answer`
- [x] Update `predict-credit-churn`
- [x] Update `predict-customer-churn`
- [x] Update `predict-household-poverty`
- [x] Update `predict-housing-prices`
- [x] Update `predict-loan-repayment`
- [x] Update `predict-malicious-cyber-connections`
- [x] Update `predict-next-purchase`
- [x] Update `predict-olympic-medals`
- [x] Update `predict-remaining-useful-life`
- [x] Update `predict-taxi-trip-duration`
- [x] Make a final scan for old references to `entity`, `entities`, `variable` and `variables`
- [x] Remove all commented out requirements that are no longer needed
- [x] Remove license files from each individual demo subdirectory

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,5 @@
 # Alteryx Open Source Demos
```
Comment (gsheni) [inline README.md:2]: Screen Shot 2021-12-14 at 3 10 09 PM

We could use this image.

Comment (thehomebrewnerd) [reply README.md:2]: @gsheni Is that image hosted somewhere already that we can pull from, or do we need to add it?

Comment (gsheni) [reply README.md:2]: Added
https://woodwork-web-images.s3.amazonaws.com/alteryx-oss-libraries.png

Comment (gsheni) [reply README.md:2]: We should center the image

File: requirements.txt [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
-TPOT==0.10.2
-tqdm>=4.19.2
-tsfresh>=0.11.1
-umap-learn>=0.3.7
-xgboost>=0.81
+# baytune>=0.2.3,<0.3.0
+# bokeh==2.4.2
+# composeml>=0.3.0
+# dask[bag]>=1.0.0
+evalml==0.39.0
```
Comment (gsheni) [inline requirements.txt:5]: I agree that we should try to minimize this as much as possible. Maybe we just need evalml and featuretools?

Comment (thehomebrewnerd) [reply requirements.txt:5]: My plan here is to pin the necessary ones, and drop everything else out. As I work through the notebooks I will add in the dependencies that become necessary until we get to the final list of what is required...and hope for no conflicts along the way.

There will be a few others that are necessary beyond the three you mention - I know `bokeh` is used by some notebooks to plot results.

File: README.md (hunk 1 above)
Comment (thehomebrewnerd) [reply README.md:2]: @gsheni Updated README.md to use this image and added the `Built at Alteryx Innovation Labs` footer like we have in other repos.

**Pull request merged**